src/Interpreters/ActionsVisitor.cpp

```diff
@@ -430,7 +430,7 @@ size_t ScopeStack::getColumnLevel(const std::string & name)
     throw Exception("Unknown identifier: " + name, ErrorCodes::UNKNOWN_IDENTIFIER);
 }
 
-void ScopeStack::addAction(const ExpressionAction & action)
+void ScopeStack::addAction(const ExpressionAction & action, bool addInput)
```

Comment:
It looks like we could make a special `addActionNoInput()` instead of this flag. It's better to make a copy-paste than add any complexity here. No one understands this code even without a flag )

P.S. coding style: `add_input` instead of `addInput`.

Reply:
>It looks like we could make a special addActionNoInput() instead of this flag. It's better to make a copy-paste than add any complexity here.

Ok

> No one understands this code even without a flag )

I thought that I was the only one )